services/fetch-wrapper.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * fetch-wrapper.ts
+ *
+ * Makes fetch() trust our CA certs. Certs loaded on first use.
+ *
+ * This is part of webOS Homebrew Channel
+ * https://github.com/webosbrew/webos-homebrew-channel
+ * Copyright 2024 throwaway96.
+ */
+
+import http from 'http';
+import https from 'https';
+
+import { loadCertDir } from './load-certs';
+
+import fetch from 'node-fetch';
+
+function createAgent(): https.Agent {
+  const certsDir = __dirname + '/certs';
+  const certs = loadCertDir(certsDir, {
+    logLevel: 0,
+  });
+
+  return new https.Agent({ ca: certs });
+}
+
+let defaultHttpsAgent: https.Agent | null = null;
+
+export function fetchWrapper(url: fetch.RequestInfo, init?: fetch.RequestInit): Promise<fetch.Response> {
+  if (defaultHttpsAgent === null) {
```

Comment:
use stylish nullish coalescing assignment operator 😎 

```ts
defaultHttpsAgent ??= createAgent();
init ??= {};
```

Reply:
The `init` one would change the semantics slightly, but I guess it doesn't matter in practice.